src/openfl/display/DisplayObjectContainer.hx

```diff
@@ -365,6 +365,38 @@ class DisplayObjectContainer extends InteractiveObject {
 	}
 	
 	
+	inline function __cleanupRemovedChildren () {
+		
+		for (orphan in __removedChildren) {
+			
+			if (orphan.stage == null) {
+				
+				orphan.__cleanup ();
+				
+			}
+			
+		}
+		
+		__removedChildren.length = 0;
```

Comment:
I have a gut feeling that using simple array for `__removedChildren` and just setting it to an empty one here would perform better than this whole Vector machinery, but this is a bit off-topic.

Reply:
I think the original thought was that `vector.length` would not resize the underlying data store, but instead set a logical length value to zero. Somewhere along the way, this didn't happen (or was removed?) so this no longer applies, but I think it might be a good idea for the future so we causing too much GC churn. 

An array + an array length value could possibly work as well, though it wouldn't be as tidy and we'd still need to zero out the values to prevent references for GC